// angular-ts-math/dist/angular-ts-math/angular-ts-math.js
export var angularMath;
(function (angularMath) {
    function getPi() { return Math.PI; }
    angularMath.getPi = getPi;
    function getE() { return Math.E; }
    angularMath.getE = getE;
    function getLn2() { return Math.LN2; }
    angularMath.getLn2 = getLn2;
    function getLn10() { return Math.LN10; }
    angularMath.getLn10 = getLn10;
    function getLog2E() { return Math.LOG2E; }
    angularMath.getLog2E = getLog2E;
    function getLog10E() { return Math.LOG10E; }
    angularMath.getLog10E = getLog10E;
    function getMinSafeInteger() { return -9007199254740991; }
    angularMath.getMinSafeInteger = getMinSafeInteger;
    function getMaxSafeInteger() { return 9007199254740991; }
    angularMath.getMaxSafeInteger = getMaxSafeInteger;
    function getRandom() { return Math.random(); }
    angularMath.getRandom = getRandom;
    function getIntegerRandomRange(min, max) { return Math.floor(Math.random() * (max - min + 1) + min); }
    angularMath.getIntegerRandomRange = getIntegerRandomRange;
    function getNIntegerRandomRange(min, max, n) {
        if (isInteger(min) && isInteger(max) && isInteger(n) && min < max && n > 0) {
            let arrayOfNumbers = [];
            for (let i = 0; i < n; i++) {
                arrayOfNumbers.push(getIntegerRandomRange(min, max));
            }
            return arrayOfNumbers;
        }
        else {
            return [];
        }
    }
    angularMath.getNIntegerRandomRange = getNIntegerRandomRange;
    function isNumber(number) { return typeof number === 'number'; }
    angularMath.isNumber = isNumber;
    function isInteger(number) { return isNumber(number) && number % 1 === 0; }
    angularMath.isInteger = isInteger;
    function isEven(number) { return isNumber(number) && number % 2 === 0; }
    angularMath.isEven = isEven;
    function isOdd(number) { return isNumber(number) && number % 2 !== 0; }
    angularMath.isOdd = isOdd;
    function isPalindrome(number) {
        if (isNumber(number)) {
            let reverseNumber = reverseOfNumber(number);
            return (isNumber(reverseNumber) && number === reverseNumber);
        }
        else {
            return false;
        }
    }
    angularMath.isPalindrome = isPalindrome;
    function numberToString(number) { return number.toString(); }
    angularMath.numberToString = numberToString;
    function numberToBinary(number) { return this.converter(number, 'B'); }
    angularMath.numberToBinary = numberToBinary;
    function numberToOct(number) { return this.converter(number, 'O'); }
    angularMath.numberToOct = numberToOct;
    function numberToHex(number) { return this.converter(number, 'H'); }
    angularMath.numberToHex = numberToHex;
    function converter(number, type) {
        if (number < 0) {
            number = 0xFFFFFFFF + number + 1;
        }
        number = parseInt(number.toString(), 10);
        switch (type) {
            case 'B':
                return number.toString(2);
            case 'O':
                return number.toString(8);
            case 'H':
                return number.toString(16);
            default:
                return number.toString(10);
        }
    }
    angularMath.converter = converter;
    function percentOfNumber(number, percent) { return (isNumber(percent) && percent >= 0) ? number * percent / 100 : 0; }
    angularMath.percentOfNumber = percentOfNumber;
    function oppositeOfNumber(number) { return -number; }
    angularMath.oppositeOfNumber = oppositeOfNumber;
    function getRest(number, div) { return number % div; }
    angularMath.getRest = getRest;
    function integerOfNumber(number) { return number.toFixed(); }
    angularMath.integerOfNumber = integerOfNumber;
    function getNumberWithDecimals(number, decimals) { return (this.isNumber(decimals) && decimals >= 0 && this.isInteger(decimals)) ? number.toFixed(decimals) : number; }
    angularMath.getNumberWithDecimals = getNumberWithDecimals;
    function absoluteOfNumber(number) { return isNumber(number) ? Math.abs(number) : 0; }
    angularMath.absoluteOfNumber = absoluteOfNumber;
    function powerOfNumber(number, p) { return Math.pow(number, p); }
    angularMath.powerOfNumber = powerOfNumber;
    function squareOfNumber(number) { return Math.sqrt(number); }
    angularMath.squareOfNumber = squareOfNumber;
    function logarithmOfNumber(number) { return Math.log(number); }
    angularMath.logarithmOfNumber = logarithmOfNumber;
    function nearOfNumber(number) { return Math.round(number); }
    angularMath.nearOfNumber = nearOfNumber;
    function nextIntegerOfNumber(number) { return Math.ceil(number); }
    angularMath.nextIntegerOfNumber = nextIntegerOfNumber;
    function backIntegerOfNumber(number) { return Math.floor(number); }
    angularMath.backIntegerOfNumber = backIntegerOfNumber;
    function reverseOfNumber(number) {
        if (isNumber(number)) {
            let stringNumber = numberToString(number) + '';
            return Number(stringNumber.split('').reverse().join(''));
        }
        else {
            return 0;
        }
    }
    angularMath.reverseOfNumber = reverseOfNumber;
    function lengthOfNumber(number) {
        if (isNumber(number)) {
            let length = 0;
            while (number > 0) {
                length++;
                number = backIntegerOfNumber(number / 10);
            }
            return length;
        }
        else {
            return 0;
        }
    }
    angularMath.lengthOfNumber = lengthOfNumber;
    function factorialOfNumber(number) {
        if (isInteger(number)) {
            let arrayOfNumbers = [];
            while (number >= 1) {
                arrayOfNumbers.push(number);
                number--;
            }
            return this.mul(arrayOfNumbers);
        }
        else {
            return 0;
        }
    }
    angularMath.factorialOfNumber = factorialOfNumber;
    function figureOfNumber(number, figure) {
        if (isNumber(number) && isInteger(number) &&
            isNumber(figure) && isInteger(figure) &&
            figure >= 0 && figure <= 9) {
            let count = 0;
            let stringNumber = numberToString(number);
            let stringFigure = numberToString(figure);
            for (let i = 0; i < stringNumber.length; i++) {
                if (stringNumber[i] === stringFigure) {
                    count++;
                }
            }
            return count;
        }
        else {
            return false;
        }
    }
    angularMath.figureOfNumber = figureOfNumber;
    function sum(...args) {
        let sum = 0;
        if (args.length > 0) {
            args.map((number) => {
                if (isNumber(number)) {
                    sum += number;
                }
            });
        }
        return sum.toFixed(2);
    }
    angularMath.sum = sum;
    function dif(...args) {
        let dif = args[0] || 0;
        if (args.length > 0) {
            args.map((number, index) => {
                if (isNumber(number) && index !== 0) {
                    dif -= number;
                }
            });
        }
        return dif.toFixed(2);
    }
    angularMath.dif = dif;
    function mul(...args) {
        let mul = 1;
        if (args.length > 0) {
            args.map(number => {
                if (isNumber(number)) {
                    mul *= number;
                }
            });
        }
        else {
            return 0;
        }
        return mul.toFixed(2);
    }
    angularMath.mul = mul;
    function div(...args) {
        let div = args[0] || 0;
        if (args.length > 0) {
            args.map((number, index) => {
                if (isNumber(number) && index !== 0) {
                    div /= number;
                }
            });
        }
        return div.toFixed(2);
    }
    angularMath.div = div;
    function getMinimum(...args) { return Math.min(...args); }
    angularMath.getMinimum = getMinimum;
    function getMaximum(...args) { return Math.max(...args); }
    angularMath.getMaximum = getMaximum;
    function cosNumber(number) { return Math.cos(number); }
    angularMath.cosNumber = cosNumber;
    function sinNumber(number) { return Math.sin(number); }
    angularMath.sinNumber = sinNumber;
    function tanNumber(number) { return Math.tan(number); }
    angularMath.tanNumber = tanNumber;
    function asinNumber(number) { return Math.asin(number); }
    angularMath.asinNumber = asinNumber;
    function acosNumber(number) { return Math.acos(number); }
    angularMath.acosNumber = acosNumber;
    function atanNumber(number) { return Math.atan(number); }
    angularMath.atanNumber = atanNumber;
    function sigma(min, max) {
        if (isInteger(min) && isInteger(max) && min < max) {
            let sum = 0;
            for (let i = min; i <= max; i++) {
                sum += i;
            }
            return sum;
        }
        else {
            return 0;
        }
    }
    angularMath.sigma = sigma;
    function pi(min, max) {
        if (isInteger(min) && isInteger(max) && min < max) {
            let mul = 1;
            for (let i = min; i <= max; i++) {
                mul *= i;
            }
            return mul;
        }
        else {
            return 0;
        }
    }
    angularMath.pi = pi;
})(angularMath || (angularMath = {}));
